packages/studio/src/components/Viewport/ViewportMenu.tsx

```diff
@@ -0,0 +1,51 @@
+import { useCallback } from "react";
+import { EXPANDED_VIEWPORTS, Viewport } from "./defaults";
+import useStudioStore from "../../store/useStudioStore";
+
+interface OptionProps extends Viewport {
+  closeMenu: () => void;
+}
+
+export default function ViewportMenu(props: {
+  closeMenu: () => void;
+}): JSX.Element {
+  return (
+    <div className="absolute z-20 rounded bg-white text-sm text-gray-700 shadow-lg flex flex-col items-start py-1">
+      {Object.values(EXPANDED_VIEWPORTS).map((val) => {
+        return <Option key={val.name} closeMenu={props.closeMenu} {...val} />;
+      })}
+    </div>
+  );
+}
+
+function Option(props: OptionProps) {
+  const [setViewportDimensions] = useStudioStore((store) => [
+    store.pagePreview.setViewportDimensions,
+  ]);
+  const { closeMenu } = props;
+  const viewportOption = props.name;
+
+  const handleSelect = useCallback(() => {
+    closeMenu();
+    setViewportDimensions({
+      name: props.name ?? "Reset Viewport",
```

Comment:
Since `OptionProps` extends `ViewPort`, won't `props.name` always be defined? It looks like all the entries in `EXPANDED_VIEWPORTS` will have a non-empty string for name.